### PR TITLE
SH-11: Homepage with slider

### PR DESCRIPTION
**Pull request by WeszeRita**
## JIRA ticket link:
https://shoperz.atlassian.net/browse/SH-11

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -15,9 +15,15 @@ import { TranslateModule } from '@ngx-translate/core';
 import { provideFirebaseApp, initializeApp } from '@angular/fire/app';
 import { getFirestore, provideFirestore } from '@angular/fire/firestore';
 import { environment } from '../environments/environment';
+import {register} from 'swiper/element/bundle';
```
Comment (krucsodaniel) [inline src/app/app.module.ts]: add space inside curly braces

File: src/app/pages/home/components/home/home.component.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<div class="flex flex-col gap-y-4">
+  <app-slider></app-slider>
+
+  <div class="flex flex-col md:flex-row w-10/12 m-auto md:border border-grey-200 p-6">
+    <div
+      *ngFor="let feature of shopFeatures; let last = last"
+      class="flex flex-row md:flex-row flex-auto justify-center items-center gap-6 md:m-auto md:w-1/5 border-none md:border-r py-4"
+      [ngClass]="{ 'border-none': last }">
+
```
Comment (krucsodaniel) [inline src/app/pages/home/components/home/home.component.html]: extra space

File: src/app/pages/home/components/home/home.component.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      <div class="w-full md:w-16 h-16">
+        <svg-icon [src]="feature.imageLink"></svg-icon>
+      </div>
+
+
+      <div class="w-full md:w-20 h-16 text-center md:text-start">
```
Comment (krucsodaniel) [inline src/app/pages/home/components/home/home.component.html]: extra space here

File: src/app/pages/home/components/home/home.component.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  <div class="flex flex-wrap justify-center items-center gap-10 w-10/12 m-auto p-10 border border-grey-200">
+    <div *ngFor="let brand of brandExpose" class="self-center">
+      <svg-icon [src]="brand"></svg-icon>
+    </div>
+  </div>
+</div>
+
+
+
```
Comment (krucsodaniel) [inline src/app/pages/home/components/home/home.component.html]: extra space

File: src/app/pages/home/home.module.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    SharedModule,
+  ],
+  declarations: [
+    SliderComponent,
+    HomeComponent,
+  ],
+   exports: [
+     SliderComponent,
+  ],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (krucsodaniel) [inline src/app/pages/home/home.module.ts]: I think exports should be the last member

File: src/shared/pipes/safe.pipe.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (krucsodaniel) [inline src/shared/pipes/safe.pipe.ts]: extra space

File: src/shared/pipes/safe.pipe.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      case 'style':
+        return this.sanitizer.bypassSecurityTrustStyle(value);
+      case 'script':
+        return this.sanitizer.bypassSecurityTrustScript(value);
+      case 'url':
+        return this.sanitizer.bypassSecurityTrustUrl(value);
+      case 'resourceUrl':
+        return this.sanitizer.bypassSecurityTrustResourceUrl(value);
+      default:
+        throw new Error(`Invalid safe type specified: ${type}`);
```
Comment (krucsodaniel) [inline src/shared/pipes/safe.pipe.ts]: add space inside template literal

**Review by krucsodaniel — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts [hunk 8]
```diff
@@ -15,9 +15,15 @@ import { TranslateModule } from '@ngx-translate/core';
...
+import { register } from 'swiper/element/bundle';
+
+register();
 
 const routes: Routes = [
-  { path: Route.base, redirectTo: Route.products, pathMatch: 'full' },
+  { path: Route.base, redirectTo: Route.home, pathMatch: 'full' },
+  { path: Route.home,
+    loadChildren: () => import('./pages/home/home.module').then(({ HomeModule }) => HomeModule),
+  },
```
Comment (bodnya29179) [inline src/app/app.module.ts]:
```suggestion
  {
    path: Route.home,
    loadChildren: () => import('./pages/home/home.module').then(({ HomeModule }) => HomeModule),
  },
```

File: src/app/pages/home/components/home/home.component.html [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<div class="flex flex-col gap-y-4">
+  <app-slider></app-slider>
+
+  <div class="flex flex-col md:flex-row w-10/12 m-auto md:border border-grey-200 p-6">
+    <div
+      *ngFor="let feature of shopFeatures; let last = last"
+      class="flex flex-row md:flex-row flex-auto justify-center items-center gap-6 md:m-auto md:w-1/5 border-none md:border-r py-4"
+      [ngClass]="{ 'border-none': last }">
```
Comment (bodnya29179) [inline src/app/pages/home/components/home/home.component.html]: also cool way for doing it: `[class.border-none]="last"`

**Review by krucsodaniel — DISMISSED** [review]
Good job, approved! :)

File: src/app/pages/home/components/home/home.component.ts [hunk 10]
```diff
@@ -0,0 +1,47 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HomeComponent {
+  readonly shopFeatures = [
```
Comment (bodnya29179) [inline src/app/pages/home/components/home/home.component.ts]: where is the interface? :)

File: src/app/pages/home/components/home/home.component.ts [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    {
+      imageLink: 'brands',
+      translationKeyLink: 'shopFeatures',
+      titleRelativeKey: 'brands',
+      detailsRelativeKey: 'brandsText',
+    }
+  ];
+
+  brandExpose = ['gama', 'technova', 'indiezone', 'tracic', 'gomezbuzz', 'shopania', 'imade', 'upside', 'wofact', 'gamerzone'];
```
Comment (bodnya29179) [inline src/app/pages/home/components/home/home.component.ts]: readonly

File: src/app/pages/home/components/slider/slider.component.html [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+          class="p-3 w-fit rounded-full bg-blue-600 text-white hover:bg-blue-500"
+          [routerLink]="['/', route.products]"
+          [queryParams]="slider.queryParams">
+          {{ buildTranslationKey('slider', 'buttonTitle') | translate }}
+        </a>
+      </div>
+
+      <img
+        class="absolute bottom-15% right-5% h-128 invisible md:visible"
+        [ngClass]="{ 'top-75': first }"
```
Comment (bodnya29179) [inline src/app/pages/home/components/slider/slider.component.html]: and here too

File: src/app/pages/home/components/slider/slider.component.ts [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+@Component({
+  selector: 'app-slider',
+  templateUrl: './slider.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SliderComponent {
+  protected readonly route = Route;
+  protected readonly sliders = [
+    {
+      class: 'w-ful h-192 m-auto bg-gradient-to-l from-turquoise-200 to-white relative',
```
Comment (bodnya29179) [inline src/app/pages/home/components/slider/slider.component.ts]: w-full and h-192 are duplicates, let's move them to template

File: src/app/pages/home/components/slider/slider.component.ts [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+import { Route, ProductCategory } from '@shared-module';
+
+@Component({
+  selector: 'app-slider',
+  templateUrl: './slider.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SliderComponent {
+  protected readonly route = Route;
+  protected readonly sliders = [
```
Comment (bodnya29179) [inline src/app/pages/home/components/slider/slider.component.ts]: interface?

File: tailwind.config.js [hunk 15]
```diff
@@ -85,6 +85,10 @@ module.exports = {
         '100': pxToRem(400),
         '128': pxToRem(512),
         '192': pxToRem(768),
+        '5%': '5%',
+        '10%': '10%',
+        '15%': '15%',
+        '20%': '20%',
```
Comment (bodnya29179) [inline tailwind.config.js]: use fractions here please :)

**Review by bodnya29179 — CHANGES_REQUESTED** [review]

File: src/app/pages/home/components/slider/slider.component.ts (hunk 13 above)
Comment (WeszeRita) [reply src/app/pages/home/components/slider/slider.component.ts]: I moved the duplicates to the template, but where should be the 'bg-gradient-to-l'? 
Is the current version ok, or it should be in the ts file to be more readable?

**Review by bodnya29179 — APPROVED** [review]

**Pull request merged**